GeosCore/drydep_mod.F90

```diff
@@ -1466,121 +1466,124 @@ SUBROUTINE DEPVEL( Input_Opt, State_Chm, State_Diag, State_Grid, &
+             !*         of light impinging the leaf; this is expressed as a
+             !*         mutliplicative factor I/(I+b) to the stomatal resistance
+             !*         where b = 50 W m-2 (equation (7) of Baldocchi et al.[1987])
+             !*       - radiative transfer of direct and diffuse radiation in the
+             !*         canopy using equations (12)-(16) from Guenther et al.[1995]
+             !*       - separate accounting of sunlit and shaded leaves using
+             !*         equation (12) of Guenther et al. [1995]
+             !*       - partitioning of the radiation at the top of the canopy into
+             !*         direct and diffuse components using a parameterization to
+             !*         results from an atmospheric radiative transfer model
+             !*         [Wang, 1996]
+             !*     The dependent variables of the function BIOFIT are the leaf
+             !*     area index (XYLAI), the cosine of zenith angle (SUNCOS) and
+             !*     the fractional cloud cover (CFRAC).  The factor GFACI
+             !*     integrates the light dependence over the canopy depth; sp even
+             !*     though RI is input per unit area of leaf it need not be scaled
+             !*     by LAI to yield a bulk canopy value because that's already
+             !*     done in the GFACI formulation.
+             !********************************************************************
+             RAD0 = RADIAT(I,J)
+             RIX = RI(LDT)
+             IF (RIX < 9999.e+0_f8) THEN
+                GFACT = 100.0e+0_f8
+                IF (TEMPC .GT. 0.e+0_f8 .AND. TEMPC .LT. 40.e+0_f8) &
+                     GFACT = 400.e+0_f8/TEMPC/(40.0e+0_f8-TEMPC)
+                GFACI = 100.e+0_f8
+                IF ( RAD0 > 0.e+0_f8 .and. XLAI(I,J,LDT) > 0.e+0_f8 ) THEN
+                   ! Now make sure all inputs to BIOFIT are flexible precision
+                   ! so that the code will compile properly (bmy, 12/18/14)
+                   XLAI_FP   = XLAI(I,J,LDT)
+                   SUNCOS_FP = SUNCOS(I,J)
+                   CFRAC_FP  = CFRAC(I,J)
+                   GFACI     = 1.e+0_f8 / BIOFIT( DRYCOEFF,  XLAI_FP, &
+                                                  SUNCOS_FP, CFRAC_FP, NN )
+                ENDIF
+
+                RIX = RIX*GFACT*GFACI
+                
+                ! Apply scaling factor to RIX when CO2 effect is turned on
+                ! (ayhwong, 6/25/19)
+                IF (Input_Opt%CO2_EFFECT) THEN
+                   RIX = RIX*Input_Opt%RS_SCALE
+                ENDIF
 
-          RIX = RIX*GFACT*GFACI
+             ENDIF   ! If RIX < 9999
 
-          ! Apply scaling factor to RIX when CO2 effect is turned on
-          ! (ayhwong, 6/25/19)
-          IF (Input_Opt%CO2_EFFECT) THEN
-             RIX = RIX*Input_Opt%RS_SCALE
-          ENDIF
+          ENDIF   ! If IUSE(I,J,IDT) > 0
```

Comment:
i think the placement of this endif is incorrect. If IUSE(I,J,LDT) > 0 then the code should to jump all of the way to line 1897 which is where the goto 170 directed to. As is, the next block of code will get executed when IUSE is 0, and that code will use RGSS and RGSO in the denominator in line 1702, resulting in divide-by-zero errors when IUSE is 0.